### PR TITLE
add client fetch transport

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/main/java/io/crate/action/sql/SQLRequest.java [hunk 1]
```diff
@@ -49,6 +50,17 @@ public SQLRequest(String stmt, Object[] args) {
     }
 
     /**
+     * Set the fetchSize. The fetchSize is the maximum number of rows which the {@link SQLResponse} will contain.
+     */
+    public void setFetchProperties(FetchProperties fetchProperties) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/action/sql/SQLRequest.java]: remove set

File: sql/src/main/java/io/crate/operation/ClientPagingReceiver.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import java.util.List;
+import java.util.Set;
+
+public class ClientPagingReceiver implements RowReceiver, CompletionListenable {
+
+    private final SettableFuture<TaskResult> resultFuture;
+
+    private CompletionListener listener = CompletionListener.NO_OP;
+    private RowUpstream rowUpstream;
+    private List<Object[]> rows = new ArrayList<>();
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/ClientPagingReceiver.java]: can we use a streambucket here?

File: sql/src/main/java/io/crate/operation/HandlerOperations.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    @Nullable
+    public ClientPagingReceiver get(UUID cursorId, TimeValue keepAlive) {
+        Context context = cursorIdToReceiverMap.get(cursorId);
+        if (context == null) {
+            return null;
+        }
+        context.lastAccess = System.nanoTime();
+        context.keepAlive = keepAlive; // TODO: should it really be possible to update the keepAlive time on each request?
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/HandlerOperations.java]: i think you can remove this todo

**Comment by dobe** [comment]
ack

File: sql/src/main/java/io/crate/action/sql/SQLRequest.java [hunk 4]
```diff
@@ -49,6 +50,17 @@ public SQLRequest(String stmt, Object[] args) {
     }
 
     /**
+     * Set the fetchSize. The fetchSize is the maximum number of rows which the {@link SQLResponse} will contain.
```
Comment (seut) [inline sql/src/main/java/io/crate/action/sql/SQLRequest.java]: pls update this comment (or remove)

**Pull request merged**